package Cmd;
import "xCmd.proto";

enum TipParam
{
  TIPPARAM_RED = 1;
  TIPPARAM_BROWSE = 2;
  TIPPARAM_ADDREDTIP = 3;
  TIPPARAM_MAX = 4;
}

enum ERedSys
{
  EREDSYS_MIN = 0;
  EREDSYS_ADD_POINT = 1;
  EREDSYS_NEW_PROFESSION = 2;
  EREDSYS_PROFESSION_UP = 3;
  EREDSYS_ROLE_IMG = 4;
  EREDSYS_MONSTER_IMG = 5;
  EREDSYS_PHOTOFRAME = 6;
  EREDSYS_SKILL_POINT = 7;

  //EREDSYS_HEAD_FASHION = 8;
  //EREDSYS_FACE_FASHION = 9;
  //EREDSYS_BACKSIDE_FASHION = 10;
  //EREDSYS_TAIL_FASHION = 11;
  //EREDSYS_CARD_WEAPON = 12;
  //EREDSYS_CARD_ACCESSORY = 13;
  //EREDSYS_CARD_ARMOUR = 14;
  //EREDSYS_CARD_BODY = 15;
  //EREDSYS_CARD_SHOES = 16;
  //EREDSYS_CARD_DECORATIONS = 17;
  //EREDSYS_CARDPIECE = 18;

  EREDSYS_TEAMAPPLY = 19;
  EREDSYS_MANUAL_MONSTER = 20;
  EREDSYS_MANUAL_HEAD = 21;
  EREDSYS_MANUAL_FACE = 22;
  EREDSYS_MANUAL_BACK = 23;
  EREDSYS_MANUAL_TAIL = 24;
  EREDSYS_MANUAL_CARD_WEAPON = 25;
  EREDSYS_MANUAL_CARD_ASSISTANT = 26;
  EREDSYS_MANUAL_CARD_ARMOUR = 27;
  EREDSYS_MANUAL_CARD_ROBE = 28;
  EREDSYS_MANUAL_CARD_SHOES = 29;
  EREDSYS_MANUAL_CARD_ACCESSORY = 30;
  EREDSYS_MANUAL_CARD_HEAD = 31;
  EREDSYS_MANUAL_NPC = 32;
  EREDSYS_MANUAL_MOUNT = 33;
  EREDSYS_SOCIAL_FRIEND_APPLY = 34;
  EREDSYS_MANUAL_MONTHCARD = 35;
  EREDSYS_MANUAL_NEWTAIL = 36;
  EREDSYS_MANUAL_COLLECTION = 37;
  EREDSYS_MANUAL_FOOD = 38;

  EREDSYS_GUILD_APPLY = 40;
  EREDSYS_GUILD_CHALLENGE_ADD = 41;
  EREDSYS_GUILD_CHALLENGE_REWARD = 42;
  EREDSYS_GUILD_ICON = 43;
  EREDSYS_PRIVATE_CHAT = 50;

  EREDSYS_TRADE_RECORD = 60;
  EREDSYS_PET_ADVENTURE = 70;

  EREDSYS_TUTOR_TUTOR_UNLOCK = 80;
  EREDSYS_TUTOR_STUDENT_UNLOCK = 81;
  EREDSYS_TUTOR_APPLY = 82;
  EREDSYS_TUTOR_TASK = 83;
  EREDSYS_TUTOR_GROW_REWARD = 85;

  EREDSYS_MANUAL_PRONTERA = 101;
  EREDSYS_MANUAL_GEFFEN = 102;
  EREDSYS_MANUAL_MOROCC = 103;
  EREDSYS_MANUAL_PAYON = 104;
  EREDSYS_MANUAL_GLAST = 105;

  EREDSYS_PACK_TEMP = 106;
  EREDSYS_AUCTION_RECORD = 84;
  EREDSYS_PEAK_LEVEL = 132;

  EREDSYS_BEING_BODY = 300;
  EREDSYS_SERVANT_RECOMMNED = 400;
  EREDSYS_SERVANT_GROWTH = 401;

  EREDSYS_DEAD_BOSS = 600;
  EREDSYS_MAX = 601;
}

enum ETipOpt
{
  ETIPOPT_MIN = 0;
  ETIPOPT_UPDATE = 1;
  ETIPOPT_DELETE = 2;
}

enum ETipItemOpt
{
  //ETIPITEMOPT_MIN = 0;
  ETIPITEMOPT_ADD = 1;
  ETIPITEMOPT_DELETE = 2;
  //ETIPITEM0PT_IGNORE = 3;
}

message RedTip
{
  optional ERedSys redsys = 1 [ default = EREDSYS_MIN ];
  optional ETipItemOpt optItem = 2 [ default = ETIPITEMOPT_ADD];
  repeated uint64 tipid = 3;
}

// scene -> client
message GameTipCmd
{
  optional Command cmd = 1 [ default = SCENE_USER_TIP_PROTOCMD ];
  optional TipParam param = 2 [ default = TIPPARAM_RED ];

  optional ETipOpt opt = 3 [ default = ETIPOPT_MIN ];
  repeated RedTip redtip = 4;
}

// client -> scene
message BrowseRedTipCmd
{
  optional Command cmd = 1 [ default = SCENE_USER_TIP_PROTOCMD ];
  optional TipParam param = 2 [ default = TIPPARAM_BROWSE ];

  optional ERedSys red = 3 [ default = EREDSYS_MIN ];
  optional uint64 tipid = 4 [ default = 0 ];
}

// client -> scene : add red tip
message AddRedTip
{
  optional Command cmd = 1 [ default = SCENE_USER_TIP_PROTOCMD ];
  optional TipParam param = 2 [ default = TIPPARAM_ADDREDTIP ];

  optional ERedSys red = 3 [ default = EREDSYS_MIN ];
  optional uint64 tipid = 4 [ default = 0 ];
}
